### PR TITLE
Feature/bc 330 create user profile

### PR DESCRIPTION
**Pull request by yeezick**
Accomplishes [BC-330](https://bootcamper.atlassian.net/jira/software/c/projects/BC/boards/2?modal=detail&selectedIssue=BC-337):
- cleans up landing page to remove user and add login button to skip user sign in flow in development
- enables the user to visit and edit their profile
- renames all API snake_case properties to camelCase
- updates interfaces to match the new user model in [backend repo](https://github.com/yeezick/bootcampr-backend/pull/5) 
- removes null acceptance from interfaces
- prepares the redux ui slice with object for auth status

### CODE REVIEW

File: src/screens/UserProfile/EditProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          <input
+            type="text"
+            name="profilePicture"
+            value={profilePicture}
+            onChange={(event) => handleInputChange(event)}
+          />
+        </label>
+
+        <label>
+          First Name
```
Comment (mendo94) [inline src/screens/UserProfile/EditProfile.tsx:48]: We didn't decide yet how we are going to layout our labels.

File: src/screens/UserProfile/UserProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { ProjectInterface } from '../../utilities/types/ProjectInterface';
+import { selectAuthUser } from '../../utilities/redux/slices/users/userSlice';
+
+export const UserProfile = () => {
+  const authUser = useSelector(selectAuthUser);
+  const navigate = useNavigate();
+
+  if (!authUser) {
```
Comment (mendo94) [inline src/screens/UserProfile/UserProfile.tsx:10]: Is this temporary? If so, you could leave a comment to remove later.

File: src/screens/UserProfile/EditProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  const authUser = useSelector(selectAuthUser);
+  const [userForm, updateUserForm] = useState<UserInterface>(authUser);
+  const params = useParams();
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const { bio, firstName, lastName, linkedinUrl, portfolioUrl, profilePicture, role, _id: userId } = userForm;
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    updateUserForm({ ...userForm, [name]: value });
```
Comment (mendo94) [inline src/screens/UserProfile/EditProfile.tsx:20]: Is this a supposed to be completed or needs to be finished in an upcoming ticket?

**Review by mendo94 — CHANGES_REQUESTED** [review]
Please review the following and let us know your thoughts behind this pr.

File: src/screens/UserProfile/EditProfile.tsx (hunk 3 above)
Comment (yeezick) [reply src/screens/UserProfile/EditProfile.tsx:20]: sorry, i'm not sure what you're referring to specifically

File: src/screens/UserProfile/UserProfile.tsx (hunk 2 above)
Comment (yeezick) [reply src/screens/UserProfile/UserProfile.tsx:10]: resolved this with a comment pointing to our jira ticket

File: src/screens/UserProfile/EditProfile.tsx (hunk 1 above)
Comment (yeezick) [reply src/screens/UserProfile/EditProfile.tsx:48]: We have not...

Comment (mendo94) [reply src/screens/UserProfile/EditProfile.tsx:48]: Well this way will work

File: src/screens/UserProfile/EditProfile.tsx (hunk 3 above)
Comment (mendo94) [reply src/screens/UserProfile/EditProfile.tsx:20]: Oh! i mean the EditProfile component, was not sure if you were just setting this screen up and getting it ready for an upcoming EditProfile ticket or it is done. I was signed in, but I was not seeing the logged in user's information being rendered in the input values on the EditProfile screen - just the initial values in the useState.

**Review by mendo94 — APPROVED** [review]
After discussing I needed to reseed the data, it does render the correct information. Thank you AMAZING work you did here, need to merge this ASAP so i can follow your structure.

**Pull request merged**